### ISSUE TITLE
[Monitor]: Hyprland crashed randomly  (Recently started happening more frequently)

### ISSUE THREAD

**Issue by niksingh710** [hyprwm/Hyprland#3191]
### Hyprland Version

Hyprland, built from branch  at commit 96d555e8e794627bfc561e294e148ab8a9961fcc dirty (). Tag: v0.29.1  flags: (if any)

### Bug or Regression?

Bug

### Description

Using it as normal it just Crashes Randomly.
More crash I noticed when I was writing a multi monitor handling script.
Not sure but a guess is this maybe causing because of some multimonitor setup?

### How to reproduce

use it and wait for crash.
Sometimes i just leave the system awake and i see it got crashed.

### Crash reports, logs, images, videos

raw [log](http://ix.io/4Fup)/[nice log](http://ix.io/4Fuq)

**Comment by niksingh710** [comment]
script had nothing special (just used to map workspaces to other monitor0

**Comment by memchr** [comment]
How much physical memory do you have? What is your average memory usage and spike?

**Comment by niksingh710** [comment]
> How much physical memory do you have? What is your average memory usage and spike?

24 gb
```bash
               total        used        free      shared  buff/cache   available
Mem:            22Gi        10Gi       3.0Gi       1.2Gi        11Gi        12Gi
Swap:             0B          0B          0B
```
Hyprland using 210MemB (report from btop)

**Comment by vaxerski** [comment]
this looks like a memory issue, either on hyprland's side or an OOM

**Comment by vaxerski** [comment]
hm, no, wait.

**Comment by vaxerski** [comment]
odd, this shouldn't happen. Can you try building wlroots and hyprland with asan?

**Comment by LDAP** [comment]
I am experiencing crashes every hour or so. Running commit 0be6b03ee972fcc4921984f3b68469a2ee121511.

Hyprland did not generate a crash report. 
Here is the core dump (without debug symbols):
```
                Stack trace of thread 197844:
                #0  0x00007f6a1c48e83c n/a (libc.so.6 + 0x8e83c)
                #1  0x00007f6a1c43e668 raise (libc.so.6 + 0x3e668)
                #2  0x00007f6a1c4264b8 abort (libc.so.6 + 0x264b8)
                #3  0x000055d94eaaed0a _Z25handleUnrecoverableSignali (Hyprland + 0xbfd0a)
                #4  0x00007f6a1c43e710 n/a (libc.so.6 + 0x3e710)
                #5  0x00007f6a1c48e83c n/a (libc.so.6 + 0x8e83c)
                #6  0x00007f6a1c43e668 raise (libc.so.6 + 0x3e668)
                #7  0x00007f6a1c4264b8 abort (libc.so.6 + 0x264b8)
                #8  0x00007f6a1c427390 n/a (libc.so.6 + 0x27390)
                #9  0x00007f6a1c51f17b __fortify_fail (libc.so.6 + 0x11f17b)
                #10 0x00007f6a1c51ec72 __fdelt_chk (libc.so.6 + 0x11ec72)
                #11 0x000055d94eb55013 _Z13hyprCtlFDTickijPv (Hyprland + 0x166013)
                #12 0x00007f6a1ce5bae2 wl_event_loop_dispatch (libwayland-server.so.0 + 0xbae2)
                #13 0x00007f6a1ce5c2d7 wl_display_run (libwayland-server.so.0 + 0xc2d7)
                #14 0x000055d94ea9ba55 main (Hyprland + 0xaca55)
                #15 0x00007f6a1c427cd0 n/a (libc.so.6 + 0x27cd0)
                #16 0x00007f6a1c427d8a __libc_start_main (libc.so.6 + 0x27d8a)
                #17 0x000055d94eaaeba5 _start (Hyprland + 0xbfba5)
```

(All core dumps  from recent crashes look the same)

**Comment by niksingh710** [comment]
i reverted to the archlinux pkg `hyprland` as i faced it on `hyprland-git` now i haven't faced any crash.

**Comment by GreenPenguino** [comment]
I am facing the same issue, also starting at Hyprland(_start+0x25).

I am running the hyprland 0.29.0-1 package on arch

**Comment by vaxerski** [comment]
ouch. Memory fault. Odd, as I try my best to remove more and more C from the project.

I'll paste instructions to build the entire wayland stack with ASan and ask one of yall to get a trace

recommended to do in tty

clone wayland (gitlab/freedesktop/wayland)
clone hyprland

wayland:
`meson ./build --prefix=/usr --buildtype=debug -Db_sanitize=address`
`sudo ninja -C build install`

hyprland:
checkout mr
`sudo make configdebug`
`sudo make debug`

edit your config to have a terminal that will launch:
`bind = SUPER, Q, exec, LD_PRELOAD=/usr/lib/libasan.so kitty`

launch hyprland from the cloned hyprland dir by doing `ASAN_OPTIONS="detect_odr_violation=0,log_path=asan.log" ./build/Hyprland`

open your term

to open any app just add `LD_PRELOAD=/usr/lib/libasan.so` to the beginning of the cmd

once you are done, to revert your horribleness of no app opening without the ld preload just go to the cloned wayland and do
`sudo rm -rf ./build`
`meson ./build --prefix=/usr --buildtype=release`
`sudo ninja -C build install`

once hyprland crashes go to `~` and look for `asan.log.xxxx`. There will be several made every instance (check their modified dates)

Zip / tar all and attach. You can see which ones belong to when by looking at the modified dates.

Once you are done, install wayland from your package manager, recompile hyprland normally (`sudo make config && make release` then manually copy the bin or just `sudo make install`)

**Comment by in0ni** [comment]
I have been having similar issues -- and mine seem to be external monitor related issues. Two issues I have just been dealing with -- but lately crashes are frequent and render my workstation unstable.

Issues I have had, and continue having:
1. When I connect a monitor _before_ I start my session (see mon_connect.txt) -- this has been the case for quite some time, so I just ensure not to plugin in monitor till after hyprland has started
2. When I lock my screen using `swaylock` at times the keyboard input does not work, if I disconnect external monitor. In order to fix this I have to reconnect/disconnect the external monitor till the keyboard receives input. Next time I'll get a output of crash.

NEW issue that is _killing me:_
1. This is recent (was about 2-4 weeks w/o updates) -- when screen wakes up after being idle (using `swayidle`) -- for now I am disabling idle...

[dpms_off_on.txt -- crash after screen goes idle](https://github.com/hyprwm/Hyprland/files/12566307/dpms_off_on.txt)
[mon_connect.txt -- crash if connect monitor before starting Hyprland](https://github.com/hyprwm/Hyprland/files/12566308/mon_connect.txt)

@vaxerski -- if I have free time, I'll give this a shot, but working weekends lately and a bit swamped (haven't even taken time for a bug report, just can't deal anymore, lol)

**Comment by LDAP** [comment]
@vaxerski Did what you said, but since compiling hyprland in debug mode and with sanitizer hyprland did not crash anymore :/.

Tried with optimization as well, didn't help to reproduce.

It's still crashing in release.

**Comment by in0ni** [comment]
> **Hyprland version**: 0.29.1-1 (arch) I have been having similar issues -- and mine seem to be external monitor related issues. 
... 
> NEW issue that is _killing me:_
> 
>     1. This is recent (was about 2-4 weeks w/o updates) -- when screen wakes up after being idle (using `swayidle`) -- for now I am disabling idle...

finally reverted to 0.28.0, which solved this for me

**Comment by niksingh710** [comment]
After recent update, I didn't face any issue like this so closing this.
